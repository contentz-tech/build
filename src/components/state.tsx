import { createContext, useContext } from "react";
import { jsx } from "@emotion/core";
import { IState } from "../state";

const StateContext = createContext<IState | null>(null);

interface StateProviderProps {
  state: IState;
  children: JSX.Element[] | JSX.Element;
}

function StateProvider({ state, children }: StateProviderProps) {
  return (
    <StateContext.Provider value={state}>{children}</StateContext.Provider>
  );
}

function useState(): IState {
  const state = useContext(StateContext);
  if (state === null)
    throw new Error(
      "Wrap your component in `StateProvider` and pass a valid state."
    );
  return state;
}

export { StateProvider, useState };
