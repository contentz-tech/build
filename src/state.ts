import { IConfig, config as getConfig } from "./getters/config";
import { articles, IArticles } from "./getters/articles";
import { pages, IPages } from "./getters/pages";
import { slides, ISlides } from "./getters/slides";
import { links, ILink } from "./getters/links";
import { statics } from "./getters/statics";
import { i18n } from "./getters/i18n";

interface IState {
  config: IConfig;
  articles: IArticles;
  pages: IPages;
  slides: ISlides;
  links: ILink[];
  statics: string[];
  i18n: any;
}

async function getState(): Promise<IState> {
  const config: IConfig = await getConfig();
  return {
    config,
    articles: await articles(config),
    pages: await pages(config),
    slides: await slides(config),
    links: await links(),
    statics: await statics(),
    i18n: await i18n(config)
  };
}

export { getState, IState };
