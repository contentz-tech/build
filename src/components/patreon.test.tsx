/** * @jest-environment jsdom */
import '@testing-library/jest-dom/extend-expect'
import { jsx } from "@emotion/core";
import { render } from "@testing-library/react";
import Patreon from "./patreon";
import { StateProvider } from "./state";

const baseState = {
  config: { },
  articles: {
    order: [],
    byPath: {}
  },
  slides: {
    order: [],
    byPath: {}
  },
  statics: [],
  pages: {
    order: [],
    byPath: {}
  },
  links: [],
  i18n: {
    patreon: {
      first: "Do you like my content?",
      link: "Become a Patreon and help me continue writing!"
    }
  }
}

describe("Patreon", () => {
  test("Without Patreon", () => {
    const { asFragment } = render(
      <StateProvider state={{ ...baseState }}>
        <Patreon />
      </StateProvider>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test("With Patreon", () => {
    const { asFragment, getByText } = render(
      <StateProvider state={{ ...baseState, config: { patreon: "sergiodxa" } }}>
        <Patreon />
      </StateProvider>
    );

    expect(asFragment()).toMatchSnapshot();
    expect(getByText(/Do you like my content?/)).toBeInTheDocument();
    expect(getByText(/Become a Patreon and help me continue writing!/))
      .toBeInTheDocument();
    expect(getByText(/Become a Patreon and help me continue writing!/))
      .toHaveAttribute("href", "https://patreon.com/sergiodxa")
  });
});
