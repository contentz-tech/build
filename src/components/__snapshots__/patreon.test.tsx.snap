// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Patreon With Patreon 1`] = `
<DocumentFragment>
  .emotion-1 {
  margin: 0;
}

.emotion-0 {
  color: #004BCC;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<p
    class="emotion-1"
  >
    Do you like my content? 
    <br />
     
    <a
      class="emotion-0"
      href="https://patreon.com/sergiodxa"
      rel="nofollow noopener"
      target="_blank"
    >
      Become a Patreon and help me continue writing!
    </a>
  </p>
</DocumentFragment>
`;

exports[`Patreon Without Patreon 1`] = `<DocumentFragment />`;
